src/libraries/Common/src/System/Security/Cryptography/CompositeMLDsa.cs

```diff
@@ -0,0 +1,1797 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Formats.Asn1;
+using System.Security.Cryptography.Asn1;
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    /// <summary>
+    ///   Represents a Composite ML-DSA key.
+    /// </summary>
+    /// <remarks>
+    ///   Developers are encouraged to program against the <see cref="CompositeMLDsa"/> base class,
+    ///   rather than any specific derived class. The derived classes are intended for interop with the underlying system
+    ///   cryptographic libraries.
+    /// </remarks>
+    [Experimental(Experimentals.PostQuantumCryptographyDiagId, UrlFormat = Experimentals.SharedUrlFormat)]
+    public abstract class CompositeMLDsa : IDisposable
+#if DESIGNTIMEINTERFACES
+#pragma warning disable SA1001
+        , IImportExportShape<CompositeMLDsa>
+#pragma warning restore SA1001
+#endif
+    {
+        private static readonly string[] s_knownOids =
+        [
+            Oids.MLDsa44WithRSA2048PssPreHashSha256,
+            Oids.MLDsa44WithRSA2048Pkcs15PreHashSha256,
+            Oids.MLDsa44WithEd25519PreHashSha512,
+            Oids.MLDsa44WithECDsaP256PreHashSha256,
+            Oids.MLDsa65WithRSA3072PssPreHashSha512,
+            Oids.MLDsa65WithRSA3072Pkcs15PreHashSha512,
+            Oids.MLDsa65WithRSA4096PssPreHashSha512,
+            Oids.MLDsa65WithRSA4096Pkcs15PreHashSha512,
+            Oids.MLDsa65WithECDsaP256PreHashSha512,
+            Oids.MLDsa65WithECDsaP384PreHashSha512,
+            Oids.MLDsa65WithECDsaBrainpoolP256r1PreHashSha512,
+            Oids.MLDsa65WithEd25519PreHashSha512,
```

Comment:
Do we care about the OIDs for things we don't support? e.g. Ed448, Ed25519, etc?

Do we want to support brainpool?

Reply:
Having all the algorithms or just the ones we support on the current platform (via `IsAlgorithmSupported`) are the two options I'd prefer. We could hardcode "which algorithms does dotnet implement on at least one platform" but I think it just adds complexity to the code.